### PR TITLE
Make the XUL panel draggable

### PR DESCRIPTION
**Pull request by 6a68**
@meandavejustice mind taking a look?

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -11,6 +11,8 @@ const launchVideo = require('./lib/launch-video');
 const sendMetricsData = require('./lib/send-metrics-data.js');
 const getDocumentDimensions = require('./lib/get-document-dimensions.js');
 const initContextMenuHandlers = require('./lib/context-menu-handlers.js');
+const { getActiveView } = require('sdk/view/core');
+const { getXULWindow } = require('sdk/window/utils');
```
Comment (pdehaan) [inline index.js]: ESLint says:
```
/home/travis/build/meandavejustice/min-vid/index.js
  15:9  error  'getXULWindow' is defined but never used               no-unused-vars
  ...
```

File: index.js [hunk 2]
```diff
@@ -25,8 +27,47 @@ const panel = require('sdk/panel').Panel({
...
 
-const { getActiveView } = require('sdk/view/core');
-getActiveView(panel).setAttribute('noautohide', true);
+// Makes an SDK panel draggable. Pass in an SDK panel.
+function draggifyPanel(panel) {
+  // Remove the panel from the XUL DOM, make some attribute changes, then
+  // reattach it. Reseating in the DOM triggers updates in the XBL bindings
+  // that give the panel draggability and remove some SDK styling.
+  let panelEl = getActiveView(panel);
+  let parentEl = panelEl.parentNode;
```
Comment (pdehaan) [inline index.js]:
```
/home/travis/build/meandavejustice/min-vid/index.js
  35:7  error  'panelEl' is never reassigned. Use 'const' instead     prefer-const
  36:7  error  'parentEl' is never reassigned. Use 'const' instead    prefer-const
 ```

File: index.js [hunk 3]
```diff
@@ -25,8 +27,47 @@ const panel = require('sdk/panel').Panel({
...
+  panelEl.removeAttribute('type');
+
+  // Next, we need a XUL document to create a drag handle. There may be better
+  // ways to obtain the document element, but this works:
+  let doc = parentEl;
+  while (doc !== null && doc.nodeType !== 9) {
+    doc = doc.parentNode;
+  }
+
+  let dragHandle = doc.createElement('label');
```
Comment (pdehaan) [inline index.js]:
```
/home/travis/build/meandavejustice/min-vid/index.js
  52:7  error  'dragHandle' is never reassigned. Use 'const' instead  prefer-const
```

**Comment by 6a68** [comment]
@pdehaan thanks :+1:

**Comment by 6a68** [comment]
If the drag handle doesn't have a background, it seems really hard to start a drag if it's been dropped outside the boundaries of the firefox window. Added a background color.

**Comment by meandavejustice** [comment]
I am going to merge this and finish up the rest of this feature in a separate branch.

**Pull request merged**